tsfc/fem.py

```diff
@@ -233,6 +234,11 @@ def flat_index(ordered_deriv):
         return result[0]
 
 
+def zero_tracking_literal(table, ufl_element):
+    track_zeros = isinstance(ufl_element, (ufl.EnrichedElement, ufl.MixedElement))
```

Comment:
So as I understand the code, the zero tracking that is added here only squeezes zeros out of the start or end of the table. e.g. if I provide an array that is `[1, 0, 0, 0, 0, 0, 0, 1]` it won't notice the zero cols in the middle.  So your only worry is if you had something like `[0, 1, 1, 1]`, you don't want squashing to `[1, 1, 1]`, right?